index.js

```diff
@@ -509,11 +519,53 @@ function uploadFiles (s3, config, files, cb, results = {done: [], errors: []}) {
   sequentially(s3, config, uploadFile, files, cb)
 }
 
+function walkSync (dir) {
+    if (!fs.lstatSync(dir).isDirectory()) return dir
+
+    return fs.readdirSync(dir).map(function(f) {
+      return walkSync(path.join(dir, f))
+    })
+}
+
+function flatten(arr) {
+  return arr.reduce(function (flat, toFlatten) {
+    return flat.concat(Array.isArray(toFlatten) ? flatten(toFlatten) : toFlatten)
+  }, [])
+}
+
+function rightExtensionForGzip(filename) {
+  return !(filename.endsWith('.js') && filename.endsWith('.html') && filename.endsWith('.css'));
+}
+
+function compressFile (filename) {
+  if (!rightExtensionForGzip(filename)) {
+    return;
+  }
+
+    var compress = zlib.createGzip(),
+        input = fs.createReadStream(filename),
+        output = fs.createWriteStream(filename + '.gz')
+
+    input.pipe(compress).pipe(output)
+    fs.unlink(filename, function() {})
+    fs.rename(filename + '.gz', filename, function() {})
+}
+
+function compressDir (dir) {
+  var files = walkSync(__dirname + '/../../' + dir)
```

Comment:
Testing your branch out, I am running into a few issues. I was trying to upload the `Tests` directory within the project to s3 just to try it out and I ran into this 
```
fs.js:946
  binding.lstat(pathModule._makeLong(path));
          ^

Error: ENOENT: no such file or directory, lstat '/Users/nickbenoit/Documents/Projects/temp/s3-website/../../test/'
    at Object.fs.lstatSync (fs.js:946:11)
    at Object.lstatSync (/Users/nickbenoit/Documents/Projects/temp/s3-website/node_modules/graceful-fs/polyfills.js:297:22)
    at walkSync (/Users/nickbenoit/Documents/Projects/temp/s3-website/index.js:523:13)
    at compressDir (/Users/nickbenoit/Documents/Projects/temp/s3-website/index.js:555:15)
    at putWebsiteContent (/Users/nickbenoit/Documents/Projects/temp/s3-website/index.js:566:5)
    at /Users/nickbenoit/Documents/Projects/temp/s3-website/s3-website.js:168:7
    at /Users/nickbenoit/Documents/Projects/temp/s3-website/index.js:498:9
    at /Users/nickbenoit/Documents/Projects/temp/s3-website/node_modules/graceful-fs/graceful-fs.js:99:16
    at /Users/nickbenoit/Documents/Projects/temp/s3-website/node_modules/graceful-fs/graceful-fs.js:43:10
    at FSReqWrap.oncomplete (fs.js:135:15)

```